File: hello-nodejs/src/client/util/VerificationModule.js
/**
 * File verification module for comparing original and downloaded files
 */

import { FileManager } from "../core/FileManager.js";

export class VerificationModule {
  constructor() {
    this.fileManager = new FileManager();
  }

  /**
   * Verify that two files are identical.
   *
   * @param {string} originalPath - Path to original file
   * @param {string} downloadedPath - Path to downloaded file
   * @returns {Promise<Object>} Verification result
   */
  async verify(originalPath, downloadedPath) {
    // Get file sizes
    const originalSize = await this.fileManager.getFileSize(originalPath);
    const downloadedSize = await this.fileManager.getFileSize(downloadedPath);

    // Calculate checksums
    const originalChecksum =
      await this.fileManager.calculateChecksum(originalPath);
    const downloadedChecksum =
      await this.fileManager.calculateChecksum(downloadedPath);

    // Compare
    const passed =
      originalSize === downloadedSize &&
      originalChecksum === downloadedChecksum;

    return {
      passed,
      originalSize,
      downloadedSize,
      originalChecksum,
      downloadedChecksum,
    };
  }
}

File: hello-csharp/src/Client/Util/VerificationModule.cs
using System.Threading.Tasks;
using AudioFileTransfer.Client.Core;

namespace AudioFileTransfer.Client.Util;

/// <summary>
/// File verification module for checking file integrity
/// </summary>
public static class VerificationModule
{
    public static async Task<VerificationResult> VerifyAsync(string originalPath, string downloadedPath)
    {
        // Compute checksums
        string originalChecksum = await FileManager.ComputeSha256Async(originalPath);
        string downloadedChecksum = await FileManager.ComputeSha256Async(downloadedPath);

        // Get file sizes
        long originalSize = await FileManager.GetFileSizeAsync(originalPath);
        long downloadedSize = await FileManager.GetFileSizeAsync(downloadedPath);

        // Compare
        bool passed = originalSize == downloadedSize && 
                     string.Equals(originalChecksum, downloadedChecksum, System.StringComparison.OrdinalIgnoreCase);

        return new VerificationResult
        {
            Passed = passed,
            OriginalSize = originalSize,
            DownloadedSize = downloadedSize,
            OriginalChecksum = originalChecksum,
            DownloadedChecksum = downloadedChecksum
        };
    }
}
